### PR TITLE
Update high_availability_settings for 24.7

### PR DESCRIPTION
**Pull request by LuminatiHD**
Closes https://github.com/puzzle/puzzle.opnsense/issues/148

### CODE REVIEW

File: plugins/modules/system_high_availability_settings.py [hunk 1]
```diff
@@ -373,30 +411,57 @@ def services_to_synchronize(
...
+            service_xml_elem = config.get("hasync").find(f"synchronize{service_id}")
+            if (
+                service_id not in sync_services
+                and service_description not in sync_services
+                and service_xml_elem is not None
+            ):
+                config.get("hasync").remove(service_xml_elem)
+
+
+def sync_compatibility(config: OPNsenseModuleConfig, compat_version: str):
```
Comment (KiLLuuuhh) [inline plugins/modules/system_high_availability_settings.py:446]: please define the return value for typing

File: plugins/module_utils/module_index.py [hunk 2]
```diff
@@ -614,4 +614,153 @@
             "configure_functions": {},
         },
     },
+    "24.7": {
```
Comment (KiLLuuuhh) [inline plugins/module_utils/module_index.py:617]: please remove this part to avoid merge issues.

**Review by KiLLuuuhh — CHANGES_REQUESTED** [review]
Some small changes but overall very sophisticated update!

File: plugins/modules/system_high_availability_settings.py [hunk 3]
```diff
@@ -37,8 +37,12 @@
     description: "If Synchronize States is enabled, it will utilize this interface for communication."
     type: str
     required: true
+  sync_compatibility:
+    description: "Newer versions of OPNsense offer additional attributes in the state synchronization, for compatibility reasons you can optionally choose an older version here. Always make sure both nodes use the same version to avoid inconsistent state tables."
+    type: str
+    required: false
```
[the remark this answers is not in the document]
Comment (LuminatiHD) [reply plugins/modules/system_high_availability_settings.py:43]: [3b9ae7d](https://github.com/puzzle/puzzle.opnsense/pull/149/commits/3b9ae7da3625f509a9afbec9f6ed0ba02dc64ae6)

**Review by KiLLuuuhh — APPROVED** [review]
Looks good to me, great work! :+1:

**Pull request merged**